src/Streamly/Internal/Data/Fold/Type.hs

```diff
@@ -815,9 +967,42 @@ teeWithMin = undefined
 -- /Unimplemented/
 --
 {-# INLINE shortest #-}
-shortest :: -- Monad m =>
+shortest ::  Monad m =>
     Fold m x a -> Fold m x a -> Fold m x a
-shortest _f1 _f2 = undefined
+shortest (Fold stepL beginL doneL) (Fold stepR beginR _) =
+    Fold step begin done
+
+    where
+
+    begin = do
+        resL <- beginL
+        resR <- beginR
+        return $
+            case resL of
+                Partial sl ->
+                    case resR of
+                        Partial sr -> Partial $ Tuple' sl sr
+                        Done br -> Done br
+                Done bl -> Done bl
+
+    step (Tuple' sL sR) a = do
+        resL <- stepL sL a
+        resR <- stepR sR a
+        return $
+            case resL of
+                Partial sL1 ->
+                    case resR of
+                        Partial sR1 -> Partial $ Tuple' sL1 sR1
+                        Done bR -> Done bR
+                Done bL -> Done bL   
+
+    done (Tuple' _ _) = undefined 
```

Comment:
We should extract the result here. Since none has finished we can take the first one.